translator/GCP_cloud_function/cloud_function/main.py

```diff
@@ -1,35 +1,53 @@
-# {'bucket': 'icone_files', 'contentType': 'image/jpeg', 'crc32c': 'ZO4N6A==', 'etag': 'CNu+up7tnu4CEAE=', 
-# 'generation': '1610745138552667', 'id': 'icone_files/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG/1610745138552667', 
-# 'kind': 'storage#object', 'md5Hash': 'yOlwCtFi0L/3GPe23rYGbw==', 
-# 'mediaLink': 'https://www.googleapis.com/download/storage/v1/b/icone_files/o/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG?generation=1610745138552667&alt=media', 
-# 'metageneration': '1', 'name': '61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG',
-#  'selfLink': 'https://www.googleapis.com/storage/v1/b/icone_files/o/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG',
-#   'size': '447421', 'storageClass': 'STANDARD', 'timeCreated': '2021-01-15T21:12:18.552Z', 'timeStorageClassUpdated': '2021-01-15T21:12:18.552Z', 
-#   'updated': '2021-01-15T21:12:18.552Z'}
-
 import icone_translator
 from gcloud import storage
 import json
 import xmltodict
+import shutil
+import urllib.request as request
+from contextlib import closing
+from google.cloud import pubsub_v1
+
+
+server = 'iconetraffic.com'
+port = '42663'
+user = 'cdot'
+password = 'icone_cdot'
+filepath = 'incidents.xml'
+
+
+ftpString = 'ftp://{0}:{1}@{2}:{3}/{4}'
+ftpUrl = ftpString.format(user, password, server, port, filepath)
+
+
+def hello_pubsub(event, context):
+
+
+  icone_data=get_ftp_file(ftpUrl)
+  icone_obj=parse_xml(icone_data)
+  info=icone_translator.initialize_info()
```

Comment:
This info object seems to be both produced and consumed by the icon_translator file. Can we just leave it in there and not fetch and repush the object?

Reply:
This was made a bit awkward because we need to set the info object to specific values for all of our work, but if another person wanted to use the translator then they would want to create their own custom info object. We should not have the initialize_info() function in the icone_translator, but should be doing that on our own. due to time constraints, we will likely only get to this next sprint.

Reply:
You could make the `info` object optional, then check if it's null on the translator side. If it's null, apply your default values produced by the `initialize_info` method.

If another application is going to call the translator, that will allow them to pass in custom values while avoiding the double call on our end here.

Reply:
great idea

Reply:
The info object is now optional